tests/components/eafm/test_config_flow.py

```diff
@@ -0,0 +1,56 @@
+"""Tests for eafm config flow."""
+import pytest
+from voluptuous.error import MultipleInvalid
+
+from homeassistant.components.eafm import const
+
+
+async def test_flow_no_discovered_stations(hass, mock_get_stations):
+    """Test config flow discovers no station."""
+    mock_get_stations.return_value = []
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": "user"}
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "no_stations"
+
+
+async def test_flow_invalid_station(hass, mock_get_stations):
+    """Test config flow errors on invalid station."""
+    mock_get_stations.return_value = [
+        {"label": "My station", "stationReference": "L12345"}
+    ]
+
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": "user"}
+    )
+    assert result["type"] == "form"
+
+    with pytest.raises(MultipleInvalid):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={"station": "My other station"}
+        )
+
+
+async def test_flow_works(hass, mock_get_stations, mock_get_station):
```

Comment:
We usually also patch `async_setup_entry` in the integration init module to avoid further set up of the integration, for the tests that create an entry.